src/core/parser.js

```diff
@@ -908,14 +908,17 @@ class Lexer {
       ch = this.nextChar();
     }
     if (ch < /* '0' = */ 0x30 || ch > /* '9' = */ 0x39) {
-      if (
-        divideBy === 10 &&
-        sign === 0 &&
-        (isWhiteSpace(ch) || ch === /* EOF = */ -1)
-      ) {
+      if (isWhiteSpace(ch) || ch === /* EOF = */ -1) {
         // This is consistent with Adobe Reader (fixes issue9252.pdf).
-        warn("Lexer.getNumber - treating a single decimal point as zero.");
-        return 0;
+        if (divideBy === 10 && sign === 0) {
```

Comment:
> Shoudn't we return `0` whatever we have ?

I'm not really sure if we want to change this "speculatively", since these code-paths only matter for *corrupt* documents and different PDF libraries probably use different fallback code-paths in these cases.
Especially for corrupt path-operators it's really quite easy to get somewhat "chaotic" rendering artefacts in corrupt documents, and I'd prefer not having to fix additional regressions here (since the original patch broken existing reference tests).

Reply:
I didn't see the original patch. Did it return `0` and break reftests? It does seem like returning 0 always seems like safe bet.

Reply:
> I didn't see the original patch. Did it return `0` and break reftests?

The changes were in https://github.com/mozilla/pdf.js/compare/36e9b7447a2a1f1062613537a2559d66d2510456..64f3dbeb489c2fd08c6918ed5c6d73d1a5788805, since otherwise existing test-cases broke.
Given that the affected `EvaluatorPreprocessor`-functionality contains a bunch of heuristics, I thus worry about additional breakage (with "chaotic" rendering) if we unconditionally start returning `0` from `Lexer.getNumber` for invalid input.